mailbox/event/json/src/main/scala/org/apache/james/event/json/MetaDataDTO.scala

```diff
@@ -48,15 +48,16 @@ object MetaDataDTO {
     val ALL_SYSTEM_FLAGS = List(ANSWERED, DELETED, DRAFT, FLAGGED, RECENT, SEEN)
 
     def toJavaFlags(serializedFlags: Array[String]): JavaMailFlags = {
-      val flagBuilder = new FlagsBuilder()
-
       serializedFlags
-        .foreach(flag => {
-          if (ALL_SYSTEM_FLAGS.contains(flag)) flagBuilder.add(stringToSystemFlag(flag))
-          else flagBuilder.add(flag)
-        })
+        .map(flag => toFlagBuilder(flag))
+        .reduceOption(_.merge(_))
```

Comment:
does  `.reduce(_ merge _)` works?

Reply:
in case of empty Flags `[]`. Reduce operation throw an exception `UnsupportedOperationException: empty.reduceLeft`, so I went to an optional one